scanpipe/models.py

```diff
@@ -1580,7 +1588,12 @@ def descendants(self):
         database query on the current CodebaseResource `path`.
         The current CodebaseResource is not included.
         """
-        return self.project.codebaseresources.filter(path__startswith=f"{self.path}/")
+        if self.path == ".":
+            return self.project.codebaseresources.exclude(path=".")
```

Comment:
This condition does not seem to be tested. Also, could you document in the dostring why the root need to be treated differently?